src/mysql_vm_helpers.py

```diff
@@ -163,6 +188,31 @@ def wait_until_mysql_connection(self) -> None:
         if not os.path.exists(MYSQLD_SOCK_FILE):
             raise MySQLServiceNotRunningError()
 
+    def connect_mysql_exporter(self) -> None:
+        """Set up mysqld-exporter config options.
+
+        Raises
+            snap.SnapError: if an issue occurs during config setting or restart
+        """
+        cache = snap.SnapCache()
+        mysqld_exporter = cache[MYSQL_EXPORTER_SNAP_NAME]
+
+        try:
+            mysqld_exporter.set(
```

Comment:
I don't know how to make it happen with a snap, but the following collectors should be disabled to reduce the parasitic monitoring load.
```
collect.auto_increment.columns
collect.info_schema.tables
collect.info_schema.tablestats
collect.perf_schema.indexiowaits
collect.perf_schema.tableiowaits
collect.perf_schema.tablelocks
collect.info_schema.userstats
collect.binlog_size
collect.info_schema.processlist
```